### DISCUSSION TITLE
AttributeError: 'V1Volume' object has no attribute 'attach_to_pod'

### DISCUSSION THREAD

**Discussion by ChaunceyZ36** [apache/airflow#30063] [Q&A]
I'm running airflow 1.10.12 on k8s, while using the kubernetes pod operator , I get the error
`AttributeError: 'V1Volume' object has no attribute 'attach_to_pod'`

here is my code:
`import pendulum
from airflow import DAG
from airflow.utils import timezone
from datetime import datetime, timedelta
from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
#from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
from airflow.operators.python_operator import PythonOperator
from airflow.operators.dummy_operator import DummyOperator
from kubernetes.client import models as k8s

volume_mount = k8s.V1VolumeMount(
    name="test-pvc",
    mount_path="/test"
)

volume = k8s.V1Volume(
    name="cronjob-dei-latest-pvc-deinas",
    persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name="test-pvc"),
)

local_tz = pendulum.timezone("Asia/Shanghai")

default_args = {
    'owner': 'Chauncey',
    'depends_on_past': False,
    'start_date': datetime(2023, 1, 7, tzinfo=local_tz),
    'email': ['xxxx@xxxx.com'],
    'email_on_failure': False,
    'email_on_retry': False,
    'retries': 0,
    'retry_delay': timedelta(minutes=5)
}

dag = DAG(
    dag_id='podtest',
    default_args=default_args,
    schedule_interval='@once',
    tags=["test"],
    catchup = False
    )

start = DummyOperator(task_id='start', dag=dag)

podtest = KubernetesPodOperator(
    namespace='default',
    image="centos:latest",
    cmds=["sh", "-c", "sleep 3600"],
    volume_mounts=[volume_mount],
    volumes=[volume],
    labels={"app": "test"},
    name="pod-test",
    task_id="podtest-task",
    get_logs=True,
    in_cluster=False,
    #cluster_context=None,
    config_file="/home/airflow/.kube/latest-config",
    dag=dag,
    is_delete_operator_pod=True
)

end = DummyOperator(task_id='end', dag=dag)

start >> podtest >> end`

Does anyone know the reason? I have tried to start a pod without mount pvc , it start successfully.

**Comment by potiuk** [comment]
1.10 is end of life for 2.5 years already. And the k8s support has been rewritten few times since . you should really upgrade because even kubernetes version that it supported back then are not supported any more, so you will have many more problems like that.

Upgrade to latest Airflow 2 - this is the best advice you can get.

**Reply by ChaunceyZ36** [reply]
Got it , thanks , but do we have the temporary solution since upgrade airflow is still on planning

**Reply by potiuk** [reply]
No idea. Really the answer is to upgrade. Likely by sticking to the old version you are in the situation like this https://xkcd.com/979/ so maybe someone else have answer or maybe you find it by googling, but i would not count too much on it. By not upgrading you largely choose to solve similar problems on your own. It's a choice that maybe you have not made yourself but until you push for upgrade, it's mostly on you and resources you can find by googling.

Harsh but i am afraid inevitable. Pity, I can't help more.

**Reply by potiuk** [reply]
No idea. Really the answer is to upgrade. Likely by sticking to the old version you are in the situation like this https://xkcd.com/979/ so maybe someone else have answer or maybe you find it by googling, but i would not count too much on it. By not upgrading you largely choose to solve similar problems on your own. It's a choice that maybe you have not made yourself but until you push for upgrade, it's mostly on you and resources you can find by googling.

Harsh but i am afraid inevitable. Pity, I can't help more.
